### DISCUSSION TITLE
How to pass JVM options

### DISCUSSION THREAD

**Discussion by thejeshgn** [jbangdev/jbang#988] [Q&A]
While running a `jar` using `jbang`, I want to add `--add-opens java.base/java.lang=ALL-UNNAMED` as an option for JVM. How do I do that?

**Comment by thejeshgn** [comment]
I see that it has been under discussion here https://github.com/jbangdev/jbang/issues/418

**Comment by quintesse** [comment]
Hi @thejeshgn . Aren't those options *always* needed? In that case you could simply add `//JAVAC_OPTIONS` or `//JAVA_OPTIONS` to your source file. See https://www.jbang.dev/documentation/guide/latest/running.html#java-and-javac-options

**Reply by quintesse** [reply]
Sorry, I see now that you say `jar` so there's no source file to add options to.
In that case you can set the `JBANG_JAVA_OPTIONS`, like this: `JBANG_JAVA_OPTIONS=--add-opens java.base/java.lang=ALL-UNNAMED` jbang myjar.jar

**Reply by quintesse** [reply]
Dammit, that's also not right. Let me think some more before replying 😅

**Reply by quintesse** [reply]
So in the latest version of Jbang there is at least _a_ way to do it. You can create an alias to refer to your jar, and that alias can have options that it will pass to the JVM at runtime, eg like this: `jbang alias add --runopt="--add-opens java.base/java.lang=ALL-UNNAMED" myjar.jar`

Btw @maxandersen , that's probably weird that we can do that in an alias but not on the command line, right? Perhaps we should at least add that `--runopt` flag to `jbang run` as well. And then perhaps a `--buildopt` as well? Wdyt?

**Reply by thejeshgn** [reply]
@quintesse Thank you.

I am using version `0.78.0` and `jbang alias add --runopt="--add-opens java.base/java.lang=ALL-UNNAMED" myjar.jar` throws

`Unknown option: '--runopt=--add-opens java.base/java.lang=ALL-UNNAMED'`

**Reply by quintesse** [reply]
Indeed, because .078 isn't the latest version :-)
Upgrade to 0.79 and it'll work.
